src/engine/engine_rd.c

```diff
@@ -106,21 +106,25 @@ static inline void _ocf_read_generic_submit_miss(struct ocf_request *req)
 
 	req->cp_data = ctx_data_alloc(cache->owner,
 			BYTES_TO_PAGES(req->byte_length));
-	if (!req->cp_data)
+	if (!req->cp_data) {
+		/* If buffer allocation for backfill fails, ignore the error */
+		ocf_cache_log(cache, log_warn, "Backfill buffer allocation "
+				"error (size %u)\n",
+				req->byte_length);
 		goto err_alloc;
+	}
 
 	ret = ctx_data_mlock(cache->owner, req->cp_data);
-	if (ret)
-		goto err_alloc;
+	if (ret) {
+		ocf_cache_log(cache, log_warn, "Backfill error\n");
+		ctx_data_free(cache->owner, req->cp_data);
+		req->cp_data = NULL;
+	}
 
+err_alloc:
 	/* Submit read request to core device. */
 	ocf_submit_volume_req(&req->core->volume, req,
 			_ocf_read_generic_miss_complete);
-
-	return;
-
-err_alloc:
-	_ocf_read_generic_miss_complete(req, -OCF_ERR_NO_MEM);
 }
```

Comment:
Something's wrong in here. Previously when alloc/mlock failed we completed with error and submited request if neither failed. Now we always submit request and never complete with error?

Reply:
If the buffer allocation fails we can't backfill but we it's not a reason to fail the user's IO